universum/modules/output/output.py

```diff
@@ -34,40 +39,57 @@ def __init__(self, *args, **kwargs) -> None:
                                 teamcity_factory=self.teamcity_driver_factory,
                                 jenkins_factory=self.terminal_driver_factory,
                                 env_type=self.settings.type)
+        self.html_driver = self._create_html_driver()
 
     def log(self, line: str) -> None:
         self.driver.log(line)
+        self.html_driver.log(line)
 
     def log_external_command(self, command: str) -> None:
         self.driver.log_external_command(command)
+        self.html_driver.log_external_command(command)
 
     def open_block(self, number: str, name: str) -> None:
         self.driver.open_block(number, name)
+        self.html_driver.open_block(number, name)
 
     def close_block(self, number: str, name: str, status: str) -> None:
         self.driver.close_block(number, name, status)
+        self.html_driver.close_block(number, name, status)
 
     def report_build_status(self, status: str) -> None:
         self.driver.change_status(status)
+        self.html_driver.change_status(status)
 
     # TODO: pass build problem to the Report module
     def report_build_problem(self, problem: str) -> None:
         self.driver.report_error(problem)
+        self.html_driver.report_error(problem)
 
     def report_skipped(self, message: str) -> None:
         self.driver.report_skipped(message)
+        self.html_driver.report_skipped(message)
 
     def report_step(self, message: str, status: str) -> None:
         self.driver.report_step(message, status)
+        self.html_driver.report_step(message, status)
 
     def log_exception(self, line: str) -> None:
         self.driver.log_exception(line)
+        self.html_driver.log_exception(line)
 
     def log_stderr(self, line: str) -> None:
         self.driver.log_stderr(line)
+        self.html_driver.log_stderr(line)
 
     def log_shell_output(self, line: str) -> None:
         self.driver.log_shell_output(line)
+        self.html_driver.log_shell_output(line)
+
+    def _create_html_driver(self):
```

Comment:
Why is this function private?

Reply:
Because its only purpose is to be called once in `Output` constructor. At least for now.